semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java

```diff
@@ -261,6 +255,57 @@ public static boolean isSqlTypeBackendForOtelFormat(
     return Optional.empty();
   }
 
+  public static Optional<String> getCassandraURI(Event event) {
+    if (SpanAttributeUtils.containsAttributeKey(event, OTEL_DB_CONNECTION_STRING)) {
+      String uri =
+          StringUtils.substringAfter(
+              StringUtils.trim(
+                  SpanAttributeUtils.getStringAttribute(event, OTEL_DB_CONNECTION_STRING)),
+              ":");
+      return getHostPortFromURI(uri);
+    }
+    Optional<String> OtBackendURI = getBackendURIForOpenTracingFormat(event);
+    if (OtBackendURI.isPresent()) {
+      return OtBackendURI;
+    }
+    Optional<String> OtelBackendURI = getBackendURIForOtelFormat(event);
+    if (OtelBackendURI.isPresent()) {
+      return OtelBackendURI;
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getElasticsearchURI(Event event) {
+    if (SpanAttributeUtils.containsAttributeKey(event, OTEL_DB_CONNECTION_STRING)
+        || SpanAttributeUtils.containsAttributeKey(event, OTEL_ELASTICSEARCH_URL)) {
+      return Optional.of(
+          SpanAttributeUtils.getFirstAvailableStringAttribute(
+              event, getAttributekeysForElasticsearchUrl()));
+    }
+    Optional<String> OtBackendURI = getBackendURIForOpenTracingFormat(event);
```

Comment:
there seems to be inconsistency between this and `isElasticsearchBackend` 
here we are using both OTel and OT tags, whereas in the other method we are just using OTel attribute

Reply:
Yeah, because we don't have any OT specific tags for ES or Cassandra.